draft-barnes-acme.md

```diff
@@ -1093,6 +1093,26 @@ type (required, string):
 To validate a DNS challenge, the server queries for TXT records under the validation domain name.  If it receives a record whose contents match the token in the challenge, then the validation succeeds.  Otherwise, the validation fails.
 
 
+## Wildcard Domain Validation
+
+When the identifier being validated is a domain name, it may contain a wildcard character "\*" as its left-most label (e.g., "\*.foo.example.com"). A wildcard certificate contains a wildcard domain in a CN or subjectAltName of type DNS-ID and automatically vouches for any and all host names within the bare domain name (the FQDN to the immediate right of the wildcard label).
+
+Domain validation for issuance of a wildcard certificate proceeds as follows:
+1. The server MUST validate that the identifier contains a single wildcard character and that the wildcard character is the left-most label. For instance, "foo.\*.example.com" is invalid.
+2. The server MUST validate that the wildcard label does not fall immediately to the left of a registry-controlled or public suffix. For instance, "\*.co.uk" is invalid.
+3. The server SHOULD validate that there exists a valid DNS A, AAAA, or CNAME record for the identifier. For instance, if the identifier being validated is "\*.example.com", the following record would suffice:
```

Comment:
None of this is required by the CA/B Forum BRs, and generally can be quite difficult to setup on DNS servers. That is, wildcard DNS records are something not all DNS servers support, and have themselves been the source of some historical controversy with the DNS space.

To issue a wildcard for *.foo.example.com, you need only validate that the Applicant is authorized for foo.example.com. If they control the DNS for foo.example.com, then by DNS's hierarchical definition, they control all subdomains as well.